File: src/contests/cg/puzzle/perl/teds-compiler.pl
# 🎮 CodinGame Puzzle - teds-compiler
# https://www.codingame.com/training/easy/teds-compiler

my $line = <STDIN>;
$line =~ s/\s+$//;
my $balance = 0;
my $best = 0;
my @c = split //, $line;
for my $i (0 .. $#c) {
    if ($c[$i] eq '<') {
        $balance++;
    } else {
        $balance--;
    }
    last if $balance < 0;
    $best = $i + 1 if $balance == 0;
}
print "$best\n";

File: src/contests/cg/puzzle/php/teds-compiler.php
<?php
// 🎮 CodinGame Puzzle - teds-compiler
// https://www.codingame.com/training/easy/teds-compiler

$line = rtrim(fgets(STDIN), "\r\n");
$balance = 0;
$best = 0;
$len = strlen($line);
for ($i = 0; $i < $len; $i++) {
    if ($line[$i] === '<') {
        $balance++;
    } else {
        $balance--;
    }
    if ($balance < 0) {
        break;
    }
    if ($balance === 0) {
        $best = $i + 1;
    }
}
echo $best . "\n";
